### PR TITLE
chore: improve tests runner

### PR DESCRIPTION
**Pull request by AlekSi**
* run subtests in parallel;
* enabled race detector;
* enable codecov.io integration.

Signed-off-by: Alexey Palazhchenko <alexey.palazhchenko@talos-systems.com>

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -50,6 +50,7 @@ steps:
     pull: always
     environment:
       PLATFORM: linux/amd64,linux/arm64
+      INTEGRATION_SKIP_CLEANUP: 1 # make things a bit faster
```
Comment (AlekSi) [inline .drone.yml:53]: Not completely sure it worth it

**Comment by AlekSi** [comment]
```
https://codecov.io/upload/v4?package=bash-1.0.6&token=<hidden>&package=bash-1.0.6&token=&branch=master[...]
```

I wonder why `token` is there twice

**Review by smira — APPROVED** [review]

**Comment by smira** [comment]
```
{'detail': ErrorDetail(string='Your CI provider is not compatible with tokenless uploads, please upload using your repository token to resolve this.', code='not_found')}
```

**Comment by AlekSi** [comment]
Yeah, I added token, and I think `token=<hidden>` means that it was injected into the environment by Drone, but bash uploader decided to pass `token` twice… Will look into it.

File: .drone.yml [hunk 2]
```diff
@@ -50,6 +50,7 @@ steps:
     pull: always
     environment:
       PLATFORM: linux/amd64,linux/arm64
```
Comment (smira) [inline .drone.yml:52]: it's too late, but we should probably remove this to simplify things: we don't want to run tests on arm64 (using test built for arm64)

**Pull request merged**